stdlib/public/stubs/LibcShims.cpp

```diff
@@ -69,11 +89,27 @@ size_t _swift_stdlib_malloc_size(const void *ptr) {
 #error No malloc_size analog known for this platform/libc.
 #endif
 
-__swift_uint32_t _swift_stdlib_arc4random(void) { return arc4random(); }
+#if defined(_MSC_VER)
+static std::random_device RandomeDevice;
+static std::mt19937 MersenneRandom(RandomeDevice());
+#endif
+
+__swift_uint32_t _swift_stdlib_arc4random(void) {
+#if defined(_MSC_VER)
+  return MersenneRandom();
+#else
+  return arc4random();
+#endif
+}
 
 __swift_uint32_t
 _swift_stdlib_arc4random_uniform(__swift_uint32_t upper_bound) {
+#if defined(_MSC_VER)
+  std::uniform_int_distribution<int> RandomUniform(1, upper_bound);
+  return RandomUniform(MersenneRandom);
+#else
   return arc4random_uniform(upper_bound);
+#endif
```

Comment:
Could you choose a name instead of ```*_arc4random_*```? It will not be related to RC4 nor ```arc4random()``` no more. If the name is changed, other source ```./stdlib/private/SwiftPrivate/PRNG.swift``` will be changed together.